dhctl/pkg/preflight/check_staticinstances_with_sshcredentials.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/deckhouse/deckhouse/dhctl/pkg/util/input"
+)
+
+func (pc *Checker) CheckStaticInstancesSSH(ctx context.Context) error {
+	docs := input.YAMLSplitRegexp.Split(pc.metaConfig.ResourcesYAML, -1)
+
+	for _, doc := range docs {
+		var res map[string]interface{}
+		if err := yaml.Unmarshal([]byte(doc), &res); err != nil {
+			return fmt.Errorf("cannot unmarshal YAML: %w", err)
+		}
+
+		if res["kind"] != "StaticInstance" {
+			continue
+		}
+
+		meta := res["metadata"].(map[string]interface{})
+		spec := res["spec"].(map[string]interface{})
+
+		instanceName := meta["name"].(string)
+		address := spec["address"].(string)
+
+		credRef := spec["credentialsRef"].(map[string]interface{})
+		credName := credRef["name"].(string)
+
+		cred, err := findSSHCredentials(docs, credName)
+		if err != nil {
+			return fmt.Errorf("instance %s: %w", instanceName, err)
+		}
+
+		if err := testSSHConnection(ctx, address, cred); err != nil {
+			return fmt.Errorf(
+				"cannot connect to %s (%s:%d): %w",
+				instanceName,
+				address,
+				cred.SSHPort,
+				err,
+			)
+		}
+	}
+	return nil
+}
+
+func findSSHCredentials(docs []string, name string) (*v1alpha2.SSHCredentialsSpec, error) {
+	for _, doc := range docs {
+		var res map[string]interface{}
+		if yaml.Unmarshal([]byte(doc), &res) != nil {
+			continue
+		}
+
+		if res["kind"] != "SSHCredentials" {
+			continue
+		}
+
+		meta := res["metadata"].(map[string]interface{})
+		if meta["name"] != name {
+			continue
+		}
+
+		spec := res["spec"].(map[string]interface{})
+		user := spec["user"].(string)
+
+		var privateKey string
+		var sudoPassword string
+
+		if k, ok := spec["privateSSHKey"].(string); ok && k != "" {
+			keyBytes, err := base64.StdEncoding.DecodeString(k)
+			if err != nil {
+				return nil, fmt.Errorf(
+					"SSHCredentials %s: cannot decode privateSSHKey: %w",
+					name,
+					err,
+				)
+			}
+			privateKey = string(keyBytes)
+		}
+
+		if sp, ok := spec["sudoPasswordEncoded"].(string); ok && sp != "" {
+			passBytes, err := base64.StdEncoding.DecodeString(sp)
+			if err != nil {
+				return nil, fmt.Errorf(
+					"SSHCredentials %s: cannot decode sudoPasswordEncoded: %w",
+					name,
+					err,
+				)
+			}
+			sudoPassword = string(passBytes)
+		}
+
+		if privateKey == "" && sudoPassword == "" {
+			return nil, fmt.Errorf(
+				"SSHCredentials %s: must contain privateSSHKey or sudoPasswordEncoded",
+				name,
+			)
+		}
+
+		port := 22
+		if p, ok := spec["sshPort"]; ok {
+			port = int(p.(int64))
+		}
+
+		return &v1alpha2.SSHCredentialsSpec{
+			User:                user,
+			PrivateSSHKey:       privateKey,
+			SudoPasswordEncoded: sudoPassword,
+			SSHPort:             port,
+		}, nil
+	}
+	return nil, fmt.Errorf("SSHCredentials %s not found", name)
+}
+
+func testSSHConnection(ctx context.Context, address string, cred *v1alpha2.SSHCredentialsSpec) error {
+	sshClient, err := sshclient.NewClientFromConfig(ctx, address, cred)
```

Comment:
and better way use retry loop (with 1 or 2 attempts) or log process in this function for checking. For better observability and show user that we check StaticInstance host in this time